src/libraries/Microsoft.Extensions.FileProviders.Physical/src/PhysicalFilesWatcher.cs

```diff
@@ -175,6 +175,10 @@ private IChangeToken GetOrAddChangeToken(string pattern)
 
         internal IChangeToken GetOrAddFilePathChangeToken(string filePath)
         {
+#if NETCOREAPP
+            var absolutePath = Path.GetFullPath(filePath, _root);
+            filePath = absolutePath.Substring(_root.Length);
+#endif
```

Comment:
@eerhardt suggestion, this should work for .NET Framework too:

```suggestion
#if NETCOREAPP
            string absolutePath = Path.GetFullPath(filePath, _root);
#else
            string absolutePath = Path.GetFullPath(Path.Combine(_root, filePath));
#endif
            filePath = absolutePath.Substring(_root.Length);
```